contents/blog/mykonos-hackathon.md

```diff
@@ -0,0 +1,153 @@
+title: What we built at our Mykonos hackathon
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+author:
+  - james-temperton
+featuredImage: >-
+  https://res.cloudinary.com/dmukukwp6/image/upload/posthog.com/contents/images/blog/aruba/beach-hog.jpg
+featuredImageType: full
+category: Inside PostHog
+tags:
+  - Offsites
+---
+
+Each year, the PostHog team gets together for our all-company offsite. This year, we headed to Mykonos – not to rave, but to code. And to enjoy a cocktail or two in the sun, of course.
+
+We’re a fully-remote company, with 47 misfits spread across ten countries. Our offsites are a really important part of our culture. They’re a great chance to get to know colleagues better and the connections formed at this time bring extra energy and creativity to our asynchronous work throughout the rest of the year.
+
+Our all-company offsites are a mix of socializing, group activities, strategic sessions, Post-its, workshops, more Post-its, and, the star of the show, the annual PostHog hackathon. Everyone in the company has to pitch a couple of ideas, then we all vote on our favorites, assemble teams, and have a little over a day to go from pitch to demo. And, in some cases, our hackathon projects are ready to ship right away.
+
+Here’s what we built in Mykonos.
+
+### The presidential briefing
+
+<BuiltBy people={['Charles Cook', 'Eric Duong', 'James Hawkins', 'James Temperton']} />
+
+As PostHog grows as a company, keeping track of everything that’s happening will get harder. Yes, we [write everything down](/newsletter/remote-working) but that creates a lot of reading, and a lot of noise. The solution? An AI-generated briefing, tailored to each individual team member and their interests.
+
+The Presidential Briefing was built by scraping PRs and Issues from GitHub, along with Slack messages, and then training an LLM to understand what’s interesting and important. The bot then produces a pithy briefing that removes the noise and gives people just the information they need. 
+
+In future, we’d want to add more data sources and build it using Llama to avoid the need to send any data to external companies.
+
+### MykoLogs
+
+<BuiltBy people={["Paul D'Ambra", "Tom Owers", "Ted Kaemming"]} />
+
+MykoLogs is a logging product that integrates with the existing PostHog SDKs, bringing backend logs straight into a shiny new product on PostHog. The best part? You can link backend logs to session recordings through the user’s session ID – letting you debug what was happening on the backend during your user's API requests. MykoLogs plays nicely with all other PostHog products, meaning logs and session replay are now BFFs. Debugging has never been this breezy!
+
+### Hog
+
+<BuiltBy people={["Marius Andra", "Anirudh Pillai"]} />
+
+What’s cooler than having your own query language? Having your own general purpose programming language, of course. That’s literally what we built: PostHog’s first ever programming language, Hog. 
+
```

Comment:
I think we need an extra para in the middle here that explains a bit about why Marius wanted to build Hog. Afaik, it's something he's been thinking about for a while.

Reply:
I'm unable to summarise that. I've been trying to sell this idea for a year now, and no matter how much or little I say, people look at me like I had bad mushrooms and don't get it.

---

This is what I wrote over a week ago when we were discussing the new team... feel free to do what you want with it:

> I've been pushing for Hog because I believe a language runtime of our own will pay gigantic dividends down the line. With a strong nod to this (all time best?) HN post (there are three ways to scale: [horizontal, vertical and deep](https://news.ycombinator.com/item?id=8902739)): If this company will grow the way it's growing, one day some team/product somewhere (data warehouse, messaging, cdp, product analytics, or all) will default to taking paths 1 and 2 out of these three:
>
> 1) Let's add a trpc webassembly runtime to interpret a javascript runtime that compiles javascript into bytecode and then let it iterate over a loop and run instructions
> 2) let's add a docker service to run a sandboxed python runtime that reads python code, iterates over a loop and runs instructions
> 3) let's iterate over a loop and run instructions
>
> Path three is going deep. Paths 1 and 2 will require vertical or horizontal scaling ($$$) to keep up with 3. Path 3 does the same with less. Thing esbuild vs webpack.
>
> Our biggest threat is not an AI product doing a better job (we'd copy and outdo them in a week). No, our biggest threat is someone else becoming the "de facto" / "no brainer" all in one analytics _platform_ before we do, and outexecutes by going deeper. Everyone's trying to copy us now: everyone's doing product analytics and/or session replay and/or feature flags. The goal posts have moved.
>
> We're still ahead of the curve as we're the only ones who do SQL at this price point. We have the data warehouse. Soon everyone will do SQL, and a warehouse, and the goal posts will move yet again. They're freaking coming for us.
>
> The biggest win our users got from HogQL is this: before we couldn't build the UI fast enough compared to what the users wanted. Now we don't need to: drop down to SQL level and ask away, or pay someone to do this for you.
>
> The way our scope is growing, people will want to express more and more complex _business logic_ on top of our platform. To express this logic, we'll build more and more complex interfaces. What's the ultimate interface to express any logic? That's right, it's a programming language.
>
> Many of our future products want users to express complex flows. We'll again run out of interfaces. Hog can do for all other products what HogQL did for Product Analytics: if you need to, drop down into hacker mode (or ask a hacker/AI), and get it over with.

Reply:
thanks! have added a line below summarizing what feels like the key element of this